code/fireball/fireballs.cpp

```diff
@@ -539,7 +614,13 @@ void fireball_set_framenum(int num)
 	}
 
 	if ( fb->fireball_render_type == FIREBALL_WARP_EFFECT )	{
-		framenum = bm_get_anim_frame(fl->bitmap_id, fb->time_elapsed, 0.0f, true);
+		float new_time = fb->time_elapsed;
+		if (fd->cinematic) {
```

Comment:
Then you can also change this and the other following checks to

`if(fd->style == warp_flare_anim::CINEMATIC){`